ipatests/test_integration/test_nfs.py

```diff
@@ -0,0 +1,252 @@
+    def install(cls, mh):
+
+        tasks.install_master(cls.master, setup_dns=True)
+        clients = (cls.clients[0], cls.replicas[0], cls.replicas[1])
+        for client in clients:
+            cls.fix_resolv_conf(client, cls.master)
+            tasks.install_client(cls.master, client)
+            client.run_command(["cat", "/etc/resolv.conf"])
+
+    def cleanup(self):
+
+        nfssrv = self.clients[0]
+        nfsclt = self.replicas[0]
+        automntclt = self.replicas[1]
+
+        nfsclt.run_command(["umount", "-a", "-t", "nfs4"])
+        nfsclt.run_command(["systemctl", "stop", "rpc-gssd"])
+
+        nfssrv.run_command(["systemctl", "stop", "nfs-server"])
+        nfssrv.run_command(["systemctl", "disable", "nfs-server"])
+        nfssrv.run_command([
+            "rm", "-f", "/etc/exports.d/krbnfs.exports",
+            "/etc/exports.d/stdnfs.exports"
+        ])
+
+        nfssrv.run_command(["rm", "-rf", "/exports"])
+
+        tasks.uninstall_client(nfsclt)
+        tasks.uninstall_client(nfssrv)
+        self.master.run_command([
+            "ipa", "host-mod", automntclt.hostname,
+            "--location", "''"
+        ])
+        # not strictly necessary, but this exercises automountlocation-del
+        self.master.run_command([
+            "ipa", "automountlocation-del", "seattle"
+        ])
+        nfsclt.run_command(["systemctl", "restart", "nfs-utils"])
+        nfssrv.run_command(["systemctl", "restart", "nfs-utils"])
+        # TODO put /etc/resolv.conf back
+        tasks.uninstall_master(self.master)
+
+    def test_prepare_users(self):
+
+        users = {
+            "athena": "p",
+            "euripides": "s"
+        }
+        temp_pass = 'temppass'
+        for user in users:
+            self.master.run_command([
+                "ipa", "user-add",
+                "%s" % user, "--first", "%s" % user,
+                "--last", "%s" % users[user],
+                '--password'], stdin_text=temp_pass
+            )
+            self.master.run_command(["kdestroy", "-A"])
+            password = "Secret123"
+            user_kinit = "%s\n%s\n%s\n" % (temp_pass, password, password)
+            self.master.run_command(
+                ['kinit', user], stdin_text=user_kinit
+            )
+            self.master.run_command(["kdestroy", "-A"])
+            tasks.kinit_admin(self.master)
+
+    def test_krb5_nfsd(self):
+
+        nfssrv = self.clients[0]
+
+        # NFS keytab management
+        self.master.run_command([
+            "ipa", "service-add", "nfs/%s" % nfssrv.hostname
+        ])
+        nfssrv.run_command([
+            "ipa-getkeytab", "-p", "nfs/%s" % nfssrv.hostname,
+            "-k", "/etc/krb5.keytab"
+        ])
+        nfssrv.run_command(["systemctl", "restart", "nfs-server"])
+        nfssrv.run_command(["systemctl", "enable", "nfs-server"])
+        time.sleep(WAIT_AFTER_INSTALL)
+
+        # NFS share
+        # TODO use a better way to associate server-side exports
```

Comment:
I am afraid this TODO will be there forever. Consider implementation or delete the comment, please.

Reply:
Agreed, removed.